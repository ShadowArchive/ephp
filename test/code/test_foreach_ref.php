<?php

$data = [ "a" => [1], "b" => [2], "c" => [3] ];

foreach($data as &$val) {
    $val[1] = $val[0] * 2;
    $val[2] = $val[0] * 3;
}

var_dump($val);
var_dump($data);

