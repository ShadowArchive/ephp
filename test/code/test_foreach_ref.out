array(3) {
  [0]=>
  int(3)
  [1]=>
  int(6)
  [2]=>
  int(9)
}
array(3) {
  ["a"]=>
  array(3) {
    [0]=>
    int(1)
    [1]=>
    int(2)
    [2]=>
    int(3)
  }
  ["b"]=>
  array(3) {
    [0]=>
    int(2)
    [1]=>
    int(4)
    [2]=>
    int(6)
  }
  ["c"]=>
  &array(3) {
    [0]=>
    int(3)
    [1]=>
    int(6)
    [2]=>
    int(9)
  }
}
